suricata/update/main.py

```diff
@@ -1028,8 +1028,6 @@ def _main():
 
     rules = []
     for filename in sorted(files):
-        if not filename.endswith(".rules"):
-            continue
```

Comment:
Hi @eoinmiller-r7 !
Thanks for your first contribution to the project. :)
I think I understand your issue but maybe dropping this can create issues for cases like filemd5 and datasets. We do not want all files to be parsed as rules.
But, it would be nice to hear @jasonish 's opinion on this.

Reply:
I think we should fake out the filename if none is present here..  When we get here: https://github.com/OISF/suricata-update/blob/master/suricata/update/main.py#L222, we know we are dealing with a single, unarchived rule file. We could give it a dummy name here, the source name, or perhaps the md5, this will allow parsing to continue, plus give the cache a better filename.

We could add a forced filename option to the source, but should first handle the case where that does not exist as its not something we want to require.